homeassistant/components/plugwise/climate.py

```diff
@@ -238,62 +214,93 @@ def temperature_unit(self):
-        self._presets_list = list(self._api.get_presets(self._domain_objects))
-        self._boiler_status = self._api.get_boiler_status(self._direct_objects)
-        self._heating_status = self._api.get_heating_status(self._direct_objects)
-        self._cooling_status = self._api.get_cooling_status(self._direct_objects)
-        self._dhw_status = self._api.get_domestic_hot_water_status(self._direct_objects)
-        self._schema_names = self._api.get_schema_names(self._domain_objects)
-        self._schema_status = self._api.get_schema_state(self._domain_objects)
-        self._current_temperature = self._api.get_current_temperature(
-            self._domain_objects
-        )
-        self._thermostat_temperature = self._api.get_thermostat_temperature(
-            self._domain_objects
-        )
-        self._schedule_temperature = self._api.get_schedule_temperature(
-            self._domain_objects
-        )
-        self._boiler_temperature = self._api.get_boiler_temperature(
-            self._domain_objects
-        )
-        self._water_pressure = self._api.get_water_pressure(self._domain_objects)
+        try:
+            await self._api.set_preset(self._loc_id, preset_mode)
+            self._preset_mode = preset_mode
+            self._setpoint = self._presets.get(self._preset_mode, "none")[0]
+            self.async_write_ha_state()
+        except Smile.PlugwiseError:
+            _LOGGER.error("Error while communicating to device")
+
+    def _process_data(self):
+        """Update the data for this climate device."""
+        climate_data = self._api.get_device_data(self._dev_id)
+        heater_central_data = self._api.get_device_data(self._api.heater_id)
+
+        if "setpoint" in climate_data:
+            self._setpoint = climate_data["setpoint"]
+        if "temperature" in climate_data:
+            self._temperature = climate_data["temperature"]
+        if "schedule_temperature" in climate_data:
+            self._schedule_temp = climate_data["schedule_temperature"]
+        if "available_schedules" in climate_data:
+            self._schema_names = climate_data["available_schedules"]
+        if "selected_schedule" in climate_data:
+            self._selected_schema = climate_data["selected_schedule"]
+            if self._selected_schema is not None:
+                self._schema_status = True
+            else:
+                self._schema_status = False
+        if "last_used" in climate_data:
+            self._last_active_schema = climate_data["last_used"]
+        if "presets" in climate_data:
+            self._presets = climate_data["presets"]
+            if self._presets:
+                self._presets_list = list(self._presets)
+        if "active_preset" in climate_data:
+            self._preset_mode = climate_data["active_preset"]
+
+        if "boiler_state" in heater_central_data:
+            if heater_central_data["boiler_state"] is not None:
```

Comment:
These two checks can be combined to:
```py
if heater_central_data.get("boiler_state") is not None:
```

Reply:
Will address all of these in the child-PR, tnx!